azurelinuxagent/common/protocol/wire.py

```diff
@@ -871,40 +888,53 @@ def get_vm_settings():
                 # clear any confusion produced by the poor choice of status code.
                 if response.status == httpclient.BAD_GATEWAY:
                     error_description = "[Internal error in HostGAPlugin] {0}".format(error_description)
+
+                if 400 <= response.status <= 499:
+                    self._vm_settings_error_reporter.report_error(error_description, _VmSettingsError.ClientError)
+                elif 500 <= response.status <= 599:
+                    self._vm_settings_error_reporter.report_error(error_description, _VmSettingsError.ServerError)
+                else:
+                    self._vm_settings_error_reporter.report_error(error_description)
+
                 raise ProtocolError("GET vmSettings [correlation ID: {0}]: {1} ({2})".format(correlation_id, response.status, error_description))
 
             for h in response.getheaders():
                 if h[0].lower() == 'etag':
                     response_etag = h[1]
                     break
             else:  # since the vmSettings were updated, the response must include an etag
-                raise ProtocolError("The vmSettings do no include an Etag [correlation ID: {0}]".format(correlation_id))
+                message = "The vmSettings do no include an Etag [correlation ID: {0}]".format(correlation_id)
+                self._vm_settings_error_reporter.report_error(message)
+                raise ProtocolError(message)
 
             response_content = self.decode_config(response.read())
             vm_settings = ExtensionsGoalStateFactory.create_from_vm_settings(response_etag, response_content)
 
-            # log the HostGAPlugin version when we fetch the vmSettings for the first time
-            is_first_vm_settings = self._vm_settings_goal_state is None
-            if is_first_vm_settings:
+            # log the HostGAPlugin version
```

Comment:
@larohra this fixes the issue you pointed out in my previous PR (always logging this when vmSettings is not supported)